Adds smoke tests

jeppe-style opened the pull request:
Adds some tests to check that given tests can be submitted.

depends on PR https://github.com/benchflow/benchflow/pull/478

VincenzoFerme commented:
Fix: https://www.codacy.com/app/BenchFlow/benchflow/pullRequest?prid=757277

VincenzoFerme commented on benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/TestManagerSmokeTestsIT.java:
```diff
@@ -0,0 +1,191 @@
+          ConfigOverride.config("minio.secretKey", MINIO_SECRET_KEY),
+          ConfigOverride.config("benchFlowExperimentManager.address", "localhost"));
+
+  // needs to be subfolder of current folder for Wercker
```
Wrong comment, it is subfolder of the `target` folder

VincenzoFerme commented on benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/TestManagerSmokeTestsIT.java:
```diff
@@ -0,0 +1,191 @@
+
+  }
+
+  private void runTest(String testName, FileDataBodyPart fileDataBodyPart) {
```
How do you ensure the tests are not actually executed, but only received and correctly handled?

jeppe-style replied:
I don't understand the question. Since this is client side the test will terminate after the response is received.

VincenzoFerme replied:
They are executed, but we do not care about it, because we just check that the API respond as expected. `TestTaskSchedulerIT` tests for correct execution of tests.

VincenzoFerme commented on benchflow-test-manager/application/src/test/resources/data/definition/exhaustive_exploration/one-at-a-time/users-memory-environment/benchflow-test.yml:
```diff
@@ -52,7 +52,6 @@ sut:
     target_service:
       name: camunda
       endpoint: /engine-rest
-      sut_ready_log_check: ready
```
Place it back here and in the other cases

VincenzoFerme reviewed (changes requested):
Why is this not covered by plain Integration Tests, and guaranteed by unit tests already?

VincenzoFerme committed:
Adds smoke tests

# Conflicts:
#	benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/helpers/TestFiles.java

VincenzoFerme committed:
Fixes Interactive Rebase Errors

VincenzoFerme committed:
Merge remote-tracking branch 'origin/devel' into feature-tm-smoke-tests

# Conflicts:
#	benchflow-test-manager/application/src/test/java/cloud/benchflow/testmanager/helpers/constants/TestFiles.java

VincenzoFerme committed:
Deletes Wrong Import

VincenzoFerme committed:
Adds Back sut_ready_log_check

VincenzoFerme committed:
Fixes a Comment

VincenzoFerme commented:
We postpone Codacy fixes
